python/tests/test_rpy_diffusive_flux.py

```diff
@@ -0,0 +1,308 @@
+import numpy as np
+import pytest
+
+import flyft
+from pytest_lazyfixture import lazy_fixture
+
+
+@pytest.fixture
+def spherical_mesh_grand():
+    return flyft.state.SphericalMesh(10.,1000, "repeat") 
+
+@pytest.fixture
+def state_grand(spherical_mesh_grand):
+    return flyft.State(flyft.state.ParallelMesh(spherical_mesh_grand), ("A", ))
+
+@pytest.fixture
+def binary_state_grand(spherical_mesh_grand):
+    return flyft.State(flyft.state.ParallelMesh(spherical_mesh_grand), ("A","B"))
+
+def test_ideal(grand,ig,rpy,state_grand):
+    state = state_grand
+    ai = 0.5
+    R = state.mesh.full.L
+    
+    ig.volumes['A'] = 1.0
+    grand.ideal = ig
+    rpy.diameters['A'] = 2 * ai
+    rpy.viscosity = 1.0
+
+    # first check OK with all zeros
+    state.fields['A'][:] = 0.
+    grand.constrain('A', np.pi * R**3 * 3.0, grand.Constraint.N)
+    rpy.compute(grand,state)
+    assert np.allclose(rpy.fluxes['A'], 0.)
+
+    # same thing with all ones
+    state.fields['A'][:] = 1
+    grand.constrain('A', np.pi * R**3 * 3.0, grand.Constraint.N)
```

Comment:
This should be the volume of the mesh because the density is 1
```suggestion
    grand.constrain('A', state.mesh.full.volume(), grand.Constraint.N)
```